src/gurt/misc.c

```diff
@@ -167,7 +167,7 @@ d_rank_list_alloc(uint32_t size)
 	d_rank_list_t		*rank_list;
 	int			 i;
 
-	D_ALLOC_PTR(rank_list);
+	D_MM_ALLOC_PTR(rank_list);
```

Comment:
Rank lists are exported/visible outside of gurt/cart so there's no way of telling where else they're freed, and I've reviewed a couple of test PRs recently where rank_list was being freed by free() rather than d_rank_list_free() so I think if you make this a MM_ALLOC_PTR then it will cause problems for users.

At the very least you'd need to change the header documentation to make it clear they were not to be freed normally.

Reply:
Ohh. Yes. Unfortunately I should not touch user visible allocations, at least for now. Thanks to catch this.